ci/dependency-update/Jenkinsfile

```diff
@@ -143,68 +174,61 @@ pipeline {
 				axes {
 					axis {
 						name 'DEPENDENCY_UPDATE_NAME'
-						values params.UPDATE_JOB == 'all' ? configurations.keySet() as String[] : params.UPDATE_JOB
+						// NOTE: Remember to update the settings() method above when changing this.
+						// And also add a new choice parameter in the parameters {} section of the pipeline
+						values 'jandex3', 'orm5.6', 'orm6.2', 'orm6.3', 'orm6-in-main-code'
 					}
 				}
 				stages {
-					stage('Init') {
-						steps {
-							sh 'ci/docker-cleanup.sh'
-							dir(env.WORKSPACE_TMP + '/.m2repository') {
-								unstash name: 'original-build-result'
-							}
-							withMavenWorkspace {
-								script {
-									env[qualify('ADDITIONAL_MAVEN_ARGS')] = settings().additionalMavenArgs ?: ''
-									if (settings().onlyRunTestDependingOn) {
-										env[qualify('ADDITIONAL_MAVEN_ARGS')] += ' -pl ' + sh(script: "./ci/list-dependent-integration-tests.sh ${settings().onlyRunTestDependingOn.join(' ')}", returnStdout: true).trim()
-									}
-								}
-							}
-						}
-					}
-					stage('Build Hibernate ORM') {
+					stage('Build') {
 						when {
 							expression {
-								return params.ORM_REPOSITORY?.trim() || params.ORM_BRANCH?.trim()
+								return params.UPDATE_JOB?.trim() == 'all' || params.UPDATE_JOB?.trim() == env.DEPENDENCY_UPDATE_NAME
 							}
 						}
```

Comment:
It seems this expression is evaluated after allocating a node to the stage, which is a waste...

Adding `beforeAgent true` should fix it; see https://www.jenkins.io/doc/book/pipeline/syntax/#evaluating-when-before-entering-agent-in-a-stage